go/pkg/outerr/outerr.go

```diff
@@ -64,3 +64,39 @@ func (s *RecordingOutErr) Stdout() []byte {
 func (s *RecordingOutErr) Stderr() []byte {
 	return s.err.Bytes()
 }
+
+// outWriter is a Writer that writes to the stdout stream of an OutErr.
+type outWriter struct {
+	OutErr
+}
+
+// NewOutWriter provides an io.Writer implementation for writing to the stdout
+// stream of an OutErr.
+func NewOutWriter(oe OutErr) io.Writer {
+	return &outWriter{OutErr: oe}
+}
+
+// Write writes to the stdout stream of the OutErr. This method always succeeds,
+// returning len(p), nil.
+func (o *outWriter) Write(p []byte) (int, error) {
+	o.WriteOut(p)
+	return len(p), nil
+}
+
+// errWriter is a Writer that writes to the stderr stream of an OutErr.
+type errWriter struct {
+	OutErr
+}
+
+// NewErrWriter provides an io.Writer implementation for writing to the stderr
+// stream of an OutErr.
+func NewErrWriter(oe OutErr) io.Writer {
+	return &errWriter{OutErr: oe}
+}
+
+// Write writes to the stderr stream of the OutErr. This method always succeeds,
+// returning len(p), nil.
+func (e *errWriter) Write(p []byte) (int, error) {
+	e.WriteErr(p)
+	return len(p), nil
+}
```

Comment:
I must admit, this package makes things more complicated than needs be.. but that's a tangent.
I think this is a reasonable compromise at this point. I just have a few comments about comments in this block.

Reply:
Yeah, I agree. I was thinking `OutErr` doesn't really seem to add much, but I also didn't want to consider removing it as in-scope for this PR.